src/bin/genemichaels.rs

```diff
@@ -196,13 +232,103 @@ fn main() {
             Offable::On(_) => true,
         },
     };
+    if args.workspace {
+        let res = || -> Result<()> {
+            {
+                let inst = time::Instant::now();
 
-    fn skip(src: &str) -> bool {
-        src.lines().take(5).any(|l| l.contains("`nogenemichaels`"))
-    }
+                // bold green
+                eprint!("\x1B[1;32m");
+                eprint!("  Formatting ");
 
-    if args.files.is_empty() {
-        match es!({
+                // reset
+                eprint!("\x1B[0;22m");
+                eprintln!("workspace...");
+                let mut cargo_toml = None;
+
+                // find workspace cargo toml
+                for parent in 0.. {
+                    let dir = if parent == 0 {
+                        ".".into()
+                    } else {
+                        "../".repeat(parent)
+                    };
+                    match std::fs::read_dir(dir)?.filter_map(|f| f.ok()).find(|f| f.file_name() == "Cargo.toml") {
+                        Some(path) if
+                            cargo_manifest::Manifest::from_path(path.path())
+                                .ok()
+                                .and_then(|m| m.workspace)
```

Comment:
I think Cargo.toml files without a workspace section probably need to be handled too (just doing `src/`), also removing this check will avoid parsing the file twice.